### PR TITLE
Teasers

### LINKED ISSUE #28: Video
From what I see, all YouTube. used in hero (#11) and two col layout (#13). I'd like to experiment with an automatic placeholder image so we can load above the fold without tanking page speed and without author having to supply one. `https://i.ytimg.com/vi_webp/${videoId}/sddefault.webp`

Also any link you a YouTube video should auto-block to this block.

### PR DESCRIPTION
**Pull request by shsteimer**
Fix #28 
Fix: #13 

Test URLs:
- After: https://teasers--ups--aemsites.aem.page/drafts/shsteimer/teasers

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -48,13 +49,34 @@ function buildLinkLists(main) {
   });
 }
 
+function buildVideoBlocks(main) {
+  main.querySelectorAll('a[href^="https://www.youtube.com/watch"]:only-child, a[href^="https://youtu.be/"]:only-child').forEach((a) => {
```
Comment (fkakatie) [inline scripts/scripts.js:53]: Interesting. I don't know that I would have thought of this.

File: blocks/teaser/teaser.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+export default async function decorate(block) {
+  const imgOrVideo = block.querySelector('.img-wrapper, .video');
+  if (!imgOrVideo) return;
+
+  const mainContainer = imgOrVideo.parentElement;
+  const heading = mainContainer.querySelector('h2');
+  if (!heading) return;
+
+  const pars = mainContainer.querySelectorAll(':scope > p:not([class])');
```
Comment (fkakatie) [inline blocks/teaser/teaser.js:42]: This is SO nitpicky, but why not `ps`? It took me an embarrassing long time to figure out what par had to do with any of this. Maybe I'm stuck in golf.

File: blocks/video/video.css [hunk 3]
```diff
@@ -0,0 +1,72 @@
+.video {
+  text-align: center;
+  max-width: 800px;
+  margin: 32px auto;
+}
+
+.video[data-embed-is-loaded="false"]:not(.placeholder) {
+  /* reserve an approximate space to avoid extensive layout shifts */
```
Comment (fkakatie) [inline blocks/video/video.css:8]: Love it.

File: blocks/video/video.css [hunk 4]
```diff
@@ -0,0 +1,72 @@
+.video {
+  text-align: center;
+  max-width: 800px;
+  margin: 32px auto;
```
Comment (fkakatie) [inline blocks/video/video.css:4]: Any reason for `px` over `rem` here?

File: blocks/teaser/teaser.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  if (teaserMedia.querySelector('.video')) {
+    teaserMedia.classList.add('teaser-video-container');
+  }
+
+  const supportingContentContainer = document.createElement('div');
+  supportingContentContainer.className = 'supporting-content-container';
+  supportingContentContainer.append(...pars);
+
+  const buttonContainer = document.createElement('div');
+  buttonContainer.className = 'button-container';
```
Comment (fkakatie) [inline blocks/teaser/teaser.js:18]: Could we leverage existing `button-wrapper` for this?

**Review by fkakatie — APPROVED** [review]

File: blocks/teaser/teaser.js (hunk 2 above)
Comment (shsteimer) [reply blocks/teaser/teaser.js:42]: basically I just want the "normal" content paragraphs. so not the `p.button-wrapper`'s and not `p.img-wrapper`'s 

simply getting any `p` without a class on it, gets me that without having to specifically exclude those classes.

File: blocks/video/video.css (hunk 4 above)
Comment (shsteimer) [reply blocks/video/video.css:4]: this is all copied from block collection. I can update to rem to conform to rest of the project though

File: scripts/scripts.js (hunk 1 above)
Comment (shsteimer) [reply scripts/scripts.js:53]: I did something similar on another project and it worked well. Pretty rare for a content author to want to link to YouTube vs. embedding it, and makes creating those embed blocks easier to author.

File: blocks/teaser/teaser.js (hunk 5 above)
Comment (shsteimer) [reply blocks/teaser/teaser.js:18]: good call.

File: blocks/teaser/teaser.js (hunk 2 above)
Comment (shsteimer) [reply blocks/teaser/teaser.js:47]: I just realized I answered a different question than what you asked. I'll rename the variable.

**Pull request merged**